<launch>
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="world_name" default="$(find gazebo_ros)/worlds/empty.world"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <param name="robot_description" textfile="$(find final3)/urdf/final3.urdf"/>

  <rosparam file="$(find final3)/config/controllers_trajectory.yaml" command="load"/>

  <node name="spawn_model" pkg="gazebo_ros" type="spawn_model"
        args="-param robot_description -urdf -model final3 -x 0 -y 0 -z 0.01" />

  <node name="controller_spawner" pkg="controller_manager" type="spawner"
        args="joint_state_controller arm_controller pitch_effort_controller roll_effort_controller"
        output="screen" />
  
  <node name="haptic_feedback_node" pkg="final3" type="haptic_feedback_node.py" output="screen" />
      
</launch>